src/services/sync/errors/validators.ts

```diff
@@ -0,0 +1,41 @@
+import { SyncError } from './index'
+
+
+export function throwIfNotNumber(val: any) {
+  if (typeof val !== 'number') throw new SyncError('Sync value is not a number: ' + val);
```

Comment:
@sheldonbaker wasn't sure what to send for ErrorDetails here

Reply:
Maybe something like:
```suggestion
  if (typeof val !== 'number') throw new SyncError('Validation error', { value: typeof val, expectedValue: 'number' });
```

(that way the error messages themselves are always consistent)